client/state/sites/selectors.js

```diff
@@ -281,6 +283,24 @@ export function isSitePreviewable( state, siteId ) {
 }
 
 /**
+ * Returns true if the site can be upgraded by the user, false if the
+ * site cannot be upgraded, or null if upgrade ability cannot be
+ * determined.
+ *
+ * @param  {Object}   state  Global state tree
+ * @param  {Number}   siteId Site ID
+ * @return {?Boolean}        Whether site is upgradeable
+ */
+export function isSiteUpgradeable( state, siteId ) {
+	// Cannot determine site upgradeability if there is no current user
+	if ( ! getCurrentUserId( state ) || ! getRawSite( state, siteId ) ) {
```

Comment:
This is mostly to cover all cases for tests, the problem with this `getCurrentUserId` selector is that it depends on another independent subtree of the state so if the whole state tree is not consistent it might lead to undefined behaviors.